src/CSET/operators/transect.py

```diff
@@ -0,0 +1,172 @@
+    # Only xmin or ymin will be zero, not both (otherwise startxy and endxy the same).
+    if startxy[1] - endxy[1] == 0:
+        latslice_only = True
+    else:
+        latslice_only = False
+
+    if startxy[0] - endxy[0] == 0:
+        lonslice_only = True
+    else:
+        lonslice_only = False
+
+    # Compute minimum gap between x/y spatial coords.
+    xmin = np.min(lon_coord.points[1:] - lon_coord.points[:-1])
+    ymin = np.min(lat_coord.points[1:] - lat_coord.points[:-1])
+
+    # Depending on the transect angle relative to the grid
+    if latslice_only:
+        xpnts = np.repeat(startxy[1], int(dist_deg / ymin))
+        ypnts = np.linspace(startxy[0], endxy[0], int(dist_deg / ymin))
+        xaxis_coord = "latitude"
+    elif lonslice_only:
+        xpnts = np.linspace(startxy[1], endxy[1], int(dist_deg / xmin))
+        ypnts = np.repeat(startxy[0], int(dist_deg / xmin))
+        xaxis_coord = "longitude"
+    else:
+        # Else use the smallest grid space in x or y direction.
+        xpnts = np.linspace(startxy[1], endxy[1], int(dist_deg / np.min([xmin, ymin])))
+        ypnts = np.linspace(startxy[0], endxy[0], int(dist_deg / np.min([xmin, ymin])))
+
+        # If change in latitude larger than change in longitude:
+        if abs(startxy[0] - endxy[0]) > abs(startxy[1] - endxy[1]):
+            xaxis_coord = "latitude"
+        elif abs(startxy[0] - endxy[0]) <= abs(startxy[1] - endxy[1]):
+            xaxis_coord = "longitude"
+
+    # Create cubelist to store interpolated points along transect.
+    interpolated_cubes = iris.cube.CubeList()
+
+    # Iterate over all points along transect.
+    for i in range(0, xpnts.shape[0]):
+        logging.info("%s/%s", i + 1, xpnts.shape[0])
+
+        # Get point along transect.
+        cube_slice = cube.interpolate(
+            [(x_name, xpnts[i]), (y_name, ypnts[i])], iris.analysis.Linear()
+        )
+
+        if xaxis_coord == "latitude":
+            cube_slice.remove_coord(x_name)
+            cube_slice.remove_coord(y_name)
+            dist_coord = iris.coords.AuxCoord(
+                ypnts[i], long_name="latitude", units="degrees"
+            )
+            cube_slice.add_aux_coord(dist_coord)
+            cube_slice = iris.util.new_axis(cube_slice, scalar_coord="latitude")
+        elif xaxis_coord == "longitude":
+            cube_slice.remove_coord(x_name)
+            cube_slice.remove_coord(y_name)
+            dist_coord = iris.coords.AuxCoord(
+                xpnts[i], long_name="longitude", units="degrees"
+            )
+            cube_slice.add_aux_coord(dist_coord)
+            cube_slice = iris.util.new_axis(cube_slice, scalar_coord="longitude")
+
+        interpolated_cubes.append(cube_slice)
+
+    # Concatenate into single cube.
+    interpolated_cubes = interpolated_cubes.concatenate()
+
+    # If concatenation successful, should be cubelist with one cube left.
```

Comment:
```suggestion
    # If concatenation successful, should be CubeList with one cube left.
    assert len(interpolated_cubes) == 1
```
Add a debug check that this precondition holds.